airbyte-integrations/connectors/source-salesforce/source_salesforce/source.py

```diff
@@ -193,29 +193,52 @@ def generate_streams(
                 )
                 continue
 
-            streams.append(self._wrap_for_concurrency(config, stream, state_manager))
-        streams.append(self._wrap_for_concurrency(config, Describe(sf_api=sf_object, catalog=self.catalog), state_manager))
+            streams.append(
+                self._convert_to_concurrent_stream(config, stream, state_manager, self._initialize_cursor(config, stream, state_manager))
+            )
+        streams.append(
+            self._convert_to_concurrent_stream(
+                config,
+                Describe(sf_api=sf_object, catalog=self.catalog),
+                state_manager,
+                self._initialize_cursor(config, stream, state_manager),
+            )
+        )
         return streams
 
-    def _wrap_for_concurrency(self, config, stream, state_manager):
-        stream_slicer_cursor = None
+    def _initialize_cursor(
+        self,
+        config: Mapping[str, Any],
+        stream: Stream,
+        state_manager: ConnectorStateManager,
+    ) -> Optional[ConcurrentCursor]:
         if stream.cursor_field:
-            stream_slicer_cursor = self._create_stream_slicer_cursor(config, state_manager, stream)
-            if hasattr(stream, "set_cursor"):
-                stream.set_cursor(stream_slicer_cursor)
-        if hasattr(stream, "parent") and hasattr(stream.parent, "set_cursor"):
-            stream_slicer_cursor = self._create_stream_slicer_cursor(config, state_manager, stream)
-            stream.parent.set_cursor(stream_slicer_cursor)
-
-        if not stream_slicer_cursor or self._get_sync_mode_from_catalog(stream) == SyncMode.full_refresh:
-            cursor = FinalStateCursor(
-                stream_name=stream.name, stream_namespace=stream.namespace, message_repository=self.message_repository
+            cursor_field_key = stream.cursor_field or ""
```

Comment:
why the default value if you're already checking `stream.cursor_field` exists

Reply:
removed